dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java

```diff
@@ -0,0 +1,152 @@
+public class ProjectV2Controller extends BaseController {
+
+    @Autowired
+    private ProjectService projectService;
+
+    /**
+     * create project
+     *
+     * @param loginUser            login user
+     * @param projectCreateRequest projectCreateRequest
+     * @return ProjectResponse ProjectResponse
+     */
+    @ApiOperation(value = "create", notes = "CREATE_PROJECT_NOTES")
+    @PostMapping(consumes = {"application/json"})
+    @ResponseStatus(HttpStatus.CREATED)
+    @ApiException(CREATE_PROJECT_ERROR)
+    @AccessLogAnnotation(ignoreRequestArgs = "loginUser")
+    public Result createProject(@ApiIgnore @RequestAttribute(value = Constants.SESSION_USER) User loginUser,
+                                @RequestBody ProjectCreateRequest projectCreateRequest) {
+        Map<String, Object> result = projectService.createProject(loginUser, projectCreateRequest.getProjectName(),
+            projectCreateRequest.getDescription());
+        return new ProjectResponse(returnDataList(result));
+    }
+
+    /**
+     * update project
+     *
+     * @param loginUser        login user
+     * @param code             project code
+     * @param projectUpdateReq projectUpdateRequest
+     * @return result Result
+     */
+    @ApiOperation(value = "update", notes = "UPDATE_PROJECT_NOTES")
+    @PutMapping(value = "/{code}", consumes = {"application/json"})
+    @ResponseStatus(HttpStatus.OK)
+    @ApiException(UPDATE_PROJECT_ERROR)
+    @AccessLogAnnotation(ignoreRequestArgs = "loginUser")
+    public Result updateProject(@ApiIgnore @RequestAttribute(value = Constants.SESSION_USER) User loginUser,
+                                @PathVariable("code") Long code,
+                                @RequestBody ProjectUpdateRequest projectUpdateReq) {
+        Map<String, Object> result = projectService.update(loginUser, code, projectUpdateReq.getProjectName(),
+            projectUpdateReq.getDescription(), projectUpdateReq.getUserName());
+        return returnDataList(result);
+    }
+
+    /**
+     * query project details by project code
+     *
+     * @param loginUser login user
+     * @param code      project code
+     * @return project detail information
+     */
+    @ApiOperation(value = "queryProjectByCode", notes = "QUERY_PROJECT_BY_ID_NOTES")
+    @ApiImplicitParams({
+        @ApiImplicitParam(name = "code", value = "PROJECT_CODE", dataType = "Long", example = "123456")
+    })
+    @GetMapping(value = "/{code}", consumes = {"application/json"})
+    @ResponseStatus(HttpStatus.OK)
+    @ApiException(QUERY_PROJECT_DETAILS_BY_CODE_ERROR)
+    @AccessLogAnnotation(ignoreRequestArgs = "loginUser")
+    public Result queryProjectByCode(@ApiIgnore @RequestAttribute(value = Constants.SESSION_USER) User loginUser,
+                                     @PathVariable("code") long code) {
+        Map<String, Object> result = projectService.queryByCode(loginUser, code);
+        return new ProjectResponse(returnDataList(result));
```

Comment:
I think we should avoid using `returnDataList` since this is not a clear method.

Reply:
if avoid using returnDataList，maybe refactor service method, it is expensive.  can we call returnDataList in pojo?

Reply:
I think this is necessary, because this is one of the purposes of refactoring. The service module only needs to fine tune the returned data structure without modifying the core logic.

Reply:
rewrite projectV2Service？

Reply:
> rewrite projectV2Service？

No need. Just optimize the existing return structure and error information.